### PR TITLE
response

### PR DESCRIPTION
**Pull request by rchezrutt88**
Didn't complete destroy or complete the model...

### CODE REVIEW

File: diagnostic.js [hunk 1]
```diff
@@ -30,25 +30,53 @@ mongoose.connect('mongodb://localhost/mongoose-crud');
...
+
+
 // Success -> Print new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  Movement.find()
+  .then(movements => {
+    console.log(movements.toJSON())
```
Comment (GA-MEB) [inline diagnostic.js:51]: Almost. `movements` is an array, but the `.toJSON` method there is only available to Mongoose documents.

**Comment by rchezrutt88** [comment]
Didn't complete as in didn't even create the model.

File: diagnostic.js [hunk 2]
```diff
@@ -30,25 +30,53 @@ mongoose.connect('mongodb://localhost/mongoose-crud');
...
+  }).catch(console.error);
+};
 // Success -> Print all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id)
+  .then(movement => {
+    console.log(movement.toJSON())
```
Comment (GA-MEB) [inline diagnostic.js:60]: Missing one edge case - if no document is found with the given id, `findById` will return `null` _which is not the same thing as hitting an error_. You need to handle that case as well; if `movement` is `null`, print out "Not Found" and exit, and if it isn't, print JSON.

File: diagnostic.js [hunk 3]
```diff
@@ -30,25 +30,53 @@ mongoose.connect('mongodb://localhost/mongoose-crud');
...
+  }).catch(console.error)
+};
 // Success -> If the specified Movement exists, print it as JSON;
 //              otherwise, print "Not Found" and exit.
 // Failure -> Console.error
 
-const update = (id, field, value) => {};
+const update = (id, field, value) => {
+  Movement.findOneAndUpdate({id: id}, {$set: {field: value}})
```
Comment (GA-MEB) [inline diagnostic.js:68]: I think the signature for this is actually
`Movement.findOneAndUpdate(id, {$set : ... })`
Also, this won't work the way you think - `{field: value}` creates an object with the key 'field'.

File: diagnostic.js [hunk 4]
```diff
@@ -30,25 +30,53 @@ mongoose.connect('mongodb://localhost/mongoose-crud');
...
+  .then(movement => {
+    console.log(movement.toJSON())
```
Comment (GA-MEB) [inline diagnostic.js:70]: See previous note about checking for a `null` value in `movement`.

**Comment by GA-MEB** [comment]
3 / 7

---

**Breakdown**
-2 for no Model
-0.5 each on `index` and `update` for syntax errors
-1 for no `delete`

Ordinarily I would take off 0.5 each on `show`, `update`, and `destroy` for not checking if `movement` is null; however, since the code samples in the `mongoose-crud` repo did not do this, I won't take off any points.

**Pull request closed without merging**